language/sygus.tex

```diff
@@ -2224,6 +2292,30 @@ \subsection{Satisfying Semantic Specifications}
 which is why their definitions are explicitly given in the definition of
 $\Psi_{\alpha}$.
 
+\subsubsection{Satisfying Semantic Specifications for Optimization Queries}
+We extend the notion of semantic correctness to responses to optimization
+queries of the form:
+\[
+\paren{\optsynthkwd\mbox{ }\paren{s_1 \ldots s_j}\mbox{ } A_1 \ldots A_k}
+\]
+Recall from \cref{sec:output} that a successful response to this command
+contains:
+\begin{enumerate}
+\item A tuple of terms $(c_1, \ldots, c_j)$ corresponding to the valuation of terms $(s_1, \ldots, s_j)$, and
+\item A tuple of functions $( \lambda X_1 \ldotp t_1, \ldots, \lambda X_n \ldotp t_n)$
+corresponding to a solution for functions-to-synthesize $f_1, \ldots, f_n$.
+\end{enumerate}
+Let $\exists f_1, \ldots, f_n \ldotp\Psi$ be the synthesis conjecture
+for the current state, and let $\Psi_{o}$ be the formula:
+\[
+\bigwedge_{i=1}^{j} s_j \teq c_j
+\]
+A solution of the above form for an optimization query
+satisfies semantic specifications if only if 
+$( \lambda X_1 \ldotp t_1, \ldots, \lambda X_n \ldotp t_n)$
+satisfies the semantic specifications
+for the synthesis conjecture $\exists f_1, \ldots, f_n. \Psi \wedge \Psi_{o}$.
+
```

Comment:
This part is still missing a discussion of the $A_1, \ldots, A_k$ attributes. I am not sure how to specify semantic satisfaction criteria for them ...